docs/examples/general/data_loading/parallel_external_source.ipynb

```diff
@@ -14,12 +14,11 @@
    "metadata": {},
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #8.            self.full_iterations = len(self.files) // batch_size
</pre><p>Maybe it can duplicate random samples to make the <code>len(self.files)</code> divisible by the batch size or add a note that this is just one (dropping, the last incomplete batch) of the options and there are different strategies that the user may apply here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Reply:
<p>I've added comments to the lines that do the divisions in non-sharding/sharding examples plus a note</p><p> </p><blockquote>Here, we decided on dropping the trailing samples if the dataset size is not divisible by the number of shards or the batch size. Note that due to shuffling of the data between the epochs, different samples will be omitted in different epochs. Nonetheless, you may decide to pad the dataset and shards by repeating some samples, so that every sample is used in every epoch.</blockquote>